### ISSUE TITLE
Make use of which mouse button is pressed when drawing on labels layer.

### ISSUE THREAD

**Issue by ianhi** [napari/napari#2189]
## 🚀 Feature

When in `paint` or `erase` mode on a `Labels` layer:
1. Middle click + drag should pan (This is the behavior in gimp)
2. right click and drag should either:
    - Change the brush size
    - zoom in and out
3. scroll should zoom in and out
    - Less sure about this one, it's nice to scroll to change brush size
 
## Motivation
I am using the Labels layer to open and correct small errors in a neural net generated `cells vs not cells` segmentation mask. So I am mainly zooming in, panning around, and then changing my brush size to make small adjustments. I found it disruptive to have to remember to press space to pan/zoom and to have to move the mouse off the region of interest in order to adjust the brush size. 

## Pitch
As above.
I think this could be done entirely with modifications to this file: https://github.com/napari/napari/blob/master/napari/layers/labels/_labels_mouse_bindings.py

## Alternatives

I thought about trying to add custom mouseevent callbacks, but because vispy events don't seem to have a concept of `event.stop_propogation` like javascript events (https://vispy.org/api/vispy.app.canvas.html?highlight=event#vispy.app.canvas.MouseEvent) I can't intercept click + drag events and prevent the drawing/erasing.

However I can customize custom bindings and make use of scroll events without issue. So I have been able to successfully change the brush size with a `wheel_callback`. Though this only solves part 2 of the original motivation.

```python
        labels = viewer.add_labels(mask, name="segmentation")

        # scrolling in paint mode changes the brush size
        def brush_size_callback(layer, event):
            if labels.mode in ["paint", "erase"]:
                if event.delta[1] > 0:
                    labels.brush_size += 1
                else:
                    labels.brush_size = max(labels.brush_size - 1, 1)

        labels.mouse_wheel_callbacks.append(brush_size_callback)
```

## Additional context
Middle click to pan when in paint mode is how gimp does it.

As proposed I think all these changes are additive and thus won't disrupt anyone.

**Comment by ianhi** [comment]
A related thing, but far more disruptive change, would be change the keybindings of the various modes to be spatially arranged on the keyboard the same way that their buttons are spatially arranged.

Like so:
![image](https://user-images.githubusercontent.com/10111092/106230018-9ff03e80-61bc-11eb-998e-d1b43b5547d0.png)



however this is easy enough to achieve with custom keybindings:


```python
        @labels.bind_key("q")
        def paint_mode(viewer):
            labels.mode = "erase"

        @labels.bind_key("w")
        def paint_mode(viewer):
            labels.mode = "fill"

        @labels.bind_key("e")
        def paint_mode(viewer):
            labels.mode = "paint"

        @labels.bind_key("r")
        def paint_mode(viewer):
            labels.mode = "pick"

        @labels.bind_key("t")
        def new_cell(viewer):
            labels.selected_label = labels.data.max() + 1
```

**Comment by sofroniewn** [comment]
Hi @ianhi thanks for raising this. There's been some recent discussion about improving mouse operations in our points layer (e.g #2159) and i think it's very reasonable to think about similar enhancements to the labels layer too.

Modifying `_labels_mouse_bindings.py` seems like a good place to start. I think I'd avoid the custom mouseevent callback/ interception idea if you can, but there is some difficulty in how implementing zooming without panning right now as they are coupled on the vispy side of things. We could break this coupling - it's coming up more and more as something we might want to do, and I have some ideas for how we might want to do it.

> A related thing, but far more disruptive change, would be change the keybindings of the various modes to be spatially arranged on the keyboard the same way that their buttons are spatially arranged.

This is an interesting idea, have you seen other UIs do this? I think we'd probably want to wait on this unless we were 100% that it was exactly what we wanted, but it is an interesting idea!

**Comment by ianhi** [comment]
Hi @sofroniewn thanks for the response.

> This is an interesting idea, have you seen other UIs do this? 

Rather embarrassingly I got the idea from real time strategy video games. Hot keys are an important aspect of those games and at some point I noticed that the default hotkeys in one game were laid out spatially the same as in menu. I think this was done to make it easier for people to learn.

**Comment by ianhi** [comment]
After sleeping I realized I left out the biggest benefit that spatial organization: Having the keys close together means that you can leave a finger on each one and don't have search around the keyboard with your left hand whenever you want to switch mode (`p` is pretty far away from `e`). This is where the real efficiency gain derives from.

**Comment by sofroniewn** [comment]
> This is where the real efficiency gain derives from.

Yeah, I like it. If we are going to move in this direction though I'd want it to be as part of a big standardization push, and take into consideration all our different layers and what we'll offer to plugin developers, for example see discussions like #1321, as I worry that if we change things too many times then people will get annoyed.

We'll also be making it possible soon to hopefully be able to customize keybindings through a preferences dialog, so we might hold off on any changes like this until have support for that functionality

**Comment by psobolewskiPhD** [comment]
So scroll now always does zoom.
Brush size can be controlled via mouse thanks to https://github.com/napari/napari/pull/5753
keybinds have been reworked to use 1, 2, 3, ...

But middle-click or right-button isn't currently used.
